include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -102,7 +102,7 @@ __pattern_walk2(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _ForwardIt
         unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f}, __n, __buf1.all_view(), __buf2.all_view());
 
     if constexpr (_IsSync())
```

Comment:
`__pattern_partial_sort_copy` uses `__pattern_walk2` in a way which requires synchronousity.  It needs the copy to finish before sort is started on the copied data.    
Also, this function has a template parameter which specifies `_IsSync`, and then we ignore it by using `__deferrable_wait()`.  Should we instead selectively use the macro to to control the `_IsSync` template parameter (only when it is safe to do so)?

Reply:
@SergeyKopienko please pay attention to this.

Reply:
@danhoeflinger, @akukanov Just for note - this error hasn't been introduced in this PR.

Reply:
Fixed.